### PR TITLE
chore: adds support for HTTP middleware.

### PR DESCRIPTION
**Pull request by jcchavezs**
The logic of the http middleware is complex enough that we should provide it.

> Thank you for contributing to Coraza WAF, your effort is greatly appreciated
> Before submitting check if what you want to add to `coraza` list meets [quality standards](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#quality-standards) before sending pull request. Thanks!

**Make sure that you've checked the boxes below before you submit PR:**

- [ ] My code includes positive and negative tests.
- [ ] I have an appropriate description with correct grammar.
- [ ] I have read [Contribution guidelines](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#contribution-guidelines), [maintainers note](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#maintainers) and [Quality standard](https://github.com/github.com/corazawaf/coraza/v3sso/coraza-waf/blob/master/CONTRIBUTING.md#quality-standards).
- [ ] My code is properly linted and passes pre-commit tests.

Thanks for your contribution :heart:

### CODE REVIEW

**Review by jptosso — APPROVED** [review]
LGTM

File: http/interceptor.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+}
+
+func (i *rwInterceptor) Header() http.Header {
+	return i.w.Header()
+}
+
+// Heavily inspired in https://github.com/openzipkin/zipkin-go/blob/master/middleware/http/server.go#L218
+func (r *rwInterceptor) wrap() http.ResponseWriter { // nolint:gocyclo
+	var (
+		hj, i0 = r.w.(http.Hijacker)
```
Comment (anuraaga) [inline http/interceptor.go:52]: `a, isA :=` pattern would probably help a lot with readability of this

Comment (jcchavezs) [reply http/interceptor.go]: Good idea. Done.

File: http/http.go [hunk 2]
```diff
@@ -16,20 +16,23 @@ import (
...
+	var (
+		client string
+		cport  int
+	)
 	// IMPORTANT: Some http.Request.RemoteAddr implementations will not contain port or contain IPV6: [2001:db8::1]:8080
-	spl := strings.Split(req.RemoteAddr, ":")
-	if len(spl) > 1 {
-		client = strings.Join(spl[0:len(spl)-1], "")
-		cport, _ = strconv.Atoi(spl[len(spl)-1])
+	idx := strings.LastIndex(req.RemoteAddr, ":")
```
Comment (anuraaga) [inline http/http.go:30]: Nit, may as well use `LastIndexByte`

**Review by anuraaga — APPROVED** [review]

**Pull request merged**